docs/concepts/rewriters.md

```diff
@@ -0,0 +1,565 @@
+
+# Rewriters
+
+`bartiq` includes a set of utilities for manipulating and simplifying symbolic expressions, known as **rewriters**. This functionality is contained in the `analysis` submodule, and backend-specific rewriters can be imported directly.
+```python
+from bartiq.analysis import sympy_rewriter
+```
+The `sympy_rewriter` operates on a single expression, however we also provide a class object to operate on `bartiq` `CompiledRoutine` objects:
+```python
+from bartiq.analysis import ResourceRewriter
+```
+Here we will give an overview of the functionality currently implemented in these objects, with example usage.
+
+## Motivation
+
+As quantum algorithms increase in complexity their symbolic resource expressions similarly become more complex. For a state of the art algorithm like double factorization (find a link to put here) the resource expressions can be almost impossible to parse, due to the sheer number of terms and symbols. Making complex expressions more palatable is a primary motivation for rewriters; gaining insights into closed-form expressions for important resource quantities is vital for fault-tolerant quantum algorithm optimization and design.
```

Comment:
(find a link to put here) Ha!
I think Fig 16 from https://arxiv.org/abs/2011.03494 is a good reference to put.